pkg/entities/record.go

```diff
@@ -209,75 +158,60 @@ func (d *dataRecord) AddInfoElement(element *InfoElement, val interface{}) (uint
 		if ipv4Add := v.To4(); ipv4Add != nil {
 			ipv4Int := big.NewInt(0)
 			ipv4Int.SetBytes(ipv4Add)
-			binary.BigEndian.PutUint32(bytesToAppend, uint32(ipv4Int.Uint64()))
+			err = util.Encode(&d.buff, uint32(ipv4Int.Uint64()))
 		} else {
-			for i, b := range v {
-				bytesToAppend[i] = b
-			}
+			err = util.Encode(&d.buff, v)
 		}
 	case String:
 		v, ok := val.(string)
 		if !ok {
 			return 0, fmt.Errorf("val argument is not of type string for this element")
 		}
 		if len(v) < 255 {
-			bytesToAppend = append(bytesToAppend, byte(len(v)))
-			bytesToAppend = append(bytesToAppend, []byte(v)...)
+			err = util.Encode(&d.buff, len(v), []byte(v))
 		} else if len(v) < 65535 {
-			bytesToAppend = append(bytesToAppend, byte(255))
-
-			byteSlice := make([]byte, 2)
-			binary.BigEndian.PutUint16(byteSlice, uint16(len(v)))
-			bytesToAppend = append(bytesToAppend, byteSlice...)
-
-			bytesToAppend = append(bytesToAppend, []byte(v)...)
+			err = util.Encode(&d.buff, byte(255), uint16(len(v)), []byte(v))
 		}
 	default:
 		return 0, fmt.Errorf("API supports only valid information elements with datatypes given in RFC7011")
 	}
 
-	bytesWritten, err := d.buff.Write(bytesToAppend)
+	//bytesWritten, err := d.buff.Write(bytesToAppend)
 	if err != nil {
 		return 0, err
 	}
 
```

Comment:
were you planning to delete this?

Reply:
It is for handling all possible error from encoding. not planning to delete it.